Work items (#363)
OctopusDeploy/docs: docs/api-and-integration/metadata/jira.md

---

> title: Jira Integration

Author: ```suggestion
title: Jira Integration with Octopus Deploy
```

> description: Configuring Octopus integration to Jira.

Author: ```suggestion
description: Configuring Octopus integration with Hosted Jira.
```

---

In addition to being able to [track metadata and work item](index.md) information through your CI/CD pipeline, Octopus can also integrate with Jira to provide progress information related to the work items during deployments.

The integration to Jira is additional to the work-item tracking built in to Octopus. The configuration described below assumes you already have the work-item tracking enabled in your pipeline and packages.

## Deployment Events

To set the scene, the end goal of the integration is to provide Jira with updates on the progress of work-items through the pipeline. It will display information similar to the following when it receives these updates for the work-items:

> ![Jira Deployments](jira-deployment.png)

Author: Remove the `was this information useful/give us feedback` text/link

> As a note, _Development_ here refers to an environment type, not an environment name. The environment name in Octopus in this example was actually Azure-Dev. See below for more detail on where this becomes important.

Author: ```suggestion
Note that "Development" here refers to an environment _type_, not an environment name. The environment name in Octopus in this example was actually "Azure-Dev". See below for more detail on where this becomes important.
```

Reviewer: I've swapped out the formatting for bolding in both cases as bold tends to stand out more than italics or speech marks.

> An important thing to note is that the link in the image above will link to the deployment in Octopus. In order for Octopus to feed this Url to Jira **you must have the Server Uri configured on the Configuration/Nodes page**.

Author: ```suggestion
Another important thing to note here is that the link in the image above will link to the deployment in Octopus. In order for Octopus to feed this Url to Jira **you must have the Server Uri configured on the Configuration/Nodes page**.

Should this be a `note` callout?
```

## Octopus Connect App and the Jira Extension

> Something worth pointing out at this point is that the Jira functionality we're demonstrating here is only available in the SaaS version of Jira. There isn't a SaaS requirement on the Octopus side of the integration though, so you can use either a self-hosted or Cloud version of Octopus.

Author: This is the third "important thing called out" in a row. Needs rewording

Author: Add a link to octopus cloud

> With that in mind, the first step in the integration is to add the Octopus Connect App in the Jira marketplace.

Author: Can we add some links here?

***TODO: add a screenshot of the marketplace tile***

> To complete the installation of the app from the marketplace you must configure some security related values. This requires aligning a couple of pieces of information from the page that appears in Jira:

Author: `some security related values` feels icky. Consider `the extension to know about your Octopus Server`.

Reviewer: This has been updated to talk about the specific values.

> ![Jira ConnectApp configuration](jira-app-config.png)

Author: Less blurred out stuff, please.

Reviewer: I've swapped this out for text with the two pieces of info that are required. Jira Base URL and Octopus Secret.

> And from the **Configuration / Jira Issue Tracker** page in Octopus:

Author: ```suggestion
And from the **Configuration -> Settings -> Jira Issue Tracker** page in Octopus:
```

Reviewer: Updated

![Octopus Jira Issue Tracker configuration](octo-jira-config.png)

> You'll probably want both of these open in separate browser windows/tabs to complete the configuration. On the Jira side you need to copy the _Octopus Installation Id_ from the Octopus page, and on the Octopus side you'll need to copy your Jira instance's *Base Url* and the *Octopus Secret* that appears on the Connect App configuration page to use as the *Jira Connect App Password*.

Author: ```suggestion
You'll probably want both of these open in separate browser windows/tabs to complete the configuration. On the Jira side you need to copy the `Octopus Installation Id` from the Octopus page, and on the Octopus side you'll need to copy your Jira instance's `Base Url` and the `Octopus Secret` that appears on the Connect App configuration page to use as the `Jira Connect App Password`.
```

maybe? Not entirely convinced thats not an abuse of formatting

Reviewer: I've swapped both code/back-ticks and italics out for bold.

## Environment Settings

> The next thing to configure relates back to our earlier note on environment types VS environment names. In Octopus you'll need to map the Jira environment types for the environments that you're going to be tracking work-items for:

Author: ```suggestion
The next thing to configure relates back to the earlier note on environment types and environment names. In Octopus you need to map the Jira environment types for the environments that you're going to be tracking work-items.
```

![Octopus Environment](octo-env.png)

> The fixed list of environment types are important for the tracking in Jira and this configuration allows you the flexibility of easily mapping any existing environments to a type.

Author: Consider removing this sentence. Not sure it adds anything.
Either that, or clarify why its important.

> The environment name does get passed through to Jira, as shown in the earlier example, so the information is still available for you to see. Also note that if you are using multi-tenancy this is fully supported and what you will see in Jira is a per tenant per environment name. As an example, the following diagram illustrates an un-tenanted deployment to Dev and then a tenanted deployment to Dev of the same version (the text is truncated in the shot, the alt text for the link reads `Deploy work-items test release 0.0.5 to Dev for Tenant A`)

Author: ```suggestion
The environment name also gets passed through to Jira, as shown in the earlier example. Also note that multi-tenancy is fully supported and the a deployment will show in Jira per tenant per environment name. 

The following diagram illustrates an un-tenanted deployment to Dev and then a tenanted deployment to Dev of the same version (the text is truncated in the shot, the alt text for the link reads `Deploy work-items test release 0.0.5 to Dev for Tenant A`)
```

It's not clear why we need environment types. What do they give me? Why should I care?

For the screenshot, can we use a shorter project name, so we can see it?

![Jira Multi-Tenant progress](jira-multi-tenant.png)

## Deployment Task Log

Once the Jira issue tracker is enabled and configured you will see blocks similar to the following appear in the log during your deployments. These show the state updates Octopus is sending through to Jira, and if you expand them the details include information on the work-item IDs etc for traceability.

> The following illustrates where Octopus tried to send an _in_progress_, and then a _successful_, state change to Jira but was unable to. **This does not impact the Octopus deployment itself, it will still be considered a successful deployment.**

Author: Can we show a success one instead of a failure one?

Also, can we have these blocks only shown in verbose mode? _think this isn't currently possible - pretty sure it'd take some tweaking of the task log renderer_

Author: If it's important to mention that a failure in the jira stuff will not impact the success or a deployment, then this should be a log message in the code.

Also, it feels it should be logged as a warning, not an error.
